openvino_xai/metrics/adcc.py

```diff
@@ -0,0 +1,132 @@
+from typing import Any, Dict, List
+
+import numpy as np
+from scipy import stats as STS
+
+from openvino_xai import Task
+from openvino_xai.explainer.explainer import Explainer, ExplainMode
+from openvino_xai.explainer.explanation import Explanation
+from openvino_xai.metrics.base import BaseMetric
+
+
+class ADCC(BaseMetric):
+    """
+    Implementation of the e Average Drop-Coherence-Complexity (ADCC) metric by Poppi, Samuele, et al 2021.
+
+    References:
+        Poppi, Samuele, et al. "Revisiting the evaluation of class activation mapping for explainability:
+        A novel metric and experimental analysis." Proceedings of the IEEE/CVF Conference on
+        Computer Vision and Pattern Recognition. 2021.
+    """
+
+    def __init__(self, model, preprocess_fn, postprocess_fn, explainer=None, device_name="CPU"):
+        super().__init__(
+            model=model, preprocess_fn=preprocess_fn, postprocess_fn=postprocess_fn, device_name=device_name
+        )
+        if explainer is None:
+            self.explainer = Explainer(
+                model=model,
+                task=Task.CLASSIFICATION,
+                preprocess_fn=self.preprocess_fn,
+                explain_mode=ExplainMode.WHITEBOX,
+            )
+        else:
+            self.explainer = explainer
+
+    def average_drop(
+        self, saliency_map: np.ndarray, class_idx: int, image: np.ndarray, model_output: np.ndarray
+    ) -> float:
+        """
+        Measures the average percentage drop in confidence for the target class when the model sees only the
+        explanation map (image masked with saliency map), instead of the full image.
+        The less the better.
+        """
+        confidence_on_input = np.max(model_output)
+
+        masked_image = (image * saliency_map[:, :, None]).astype(np.uint8)
+        prediction_on_saliency_map = self.model_predict(masked_image)
+        confidence_on_saliency_map = prediction_on_saliency_map[class_idx]
+
+        return max(0.0, confidence_on_input - confidence_on_saliency_map) / confidence_on_input
+
+    def coherency(self, saliency_map: np.ndarray, class_idx: int, image: np.ndarray) -> float:
+        """
+        Measures the coherency of the saliency map. The explanation map (image masked with saliency map) should contain all the relevant features that explain a prediction and should remove useless features in a coherent way.
+        Saliency map and saliency map of exlanation map should be similar.
+        The more the better.
+        """
+
+        masked_image = image * saliency_map[:, :, None]
+        saliency_map_mapped_image = self.explainer(masked_image, targets=[class_idx], colormap=False, scaling=False)
+        saliency_map_mapped_image = saliency_map_mapped_image.saliency_map[class_idx]
+
+        A, B = saliency_map, saliency_map_mapped_image
+        # Pearson correlation coefficient
+        Asq, Bsq = A.flatten(), B.flatten()
+        y, _ = STS.pearsonr(Asq, Bsq)
```

Comment:
These are from some convention?
What does `sq` mean?
STS? Can we just import `pearsonr`?
Please ignore if you think the code readers would likely be OK.

Reply:
I follow the original implementation from here https://github.com/aimagelab/ADCC/blob/main/metrics/coherency.py

Good point with importing just `pearsonr`, I'll update that and naming in the next PR.